utils/deprecate_models.py

```diff
@@ -0,0 +1,472 @@
+                    indented_block.append(line)
+                    sub_block.append("\n".join(indented_block))
+                    indented_block = []
+                    open_indent_level = -1
+                else:
+                    # We're still in the indented block
+                    indented_block.append(line)
+
+            # We have a comment line - create a new sub-block so lines above the comment are grouped together when sorting
+            elif line.strip().startswith("#"):
+                if sub_block:
+                    else_block.append(sub_block)
+                    sub_block = []
+
+                # When adding else blocks to the file lines, we sort each sub-block, but not between sub-blocks.
+                # Comment lines are added directly to make sure they stay between the sub-blocks they're associated with
+                # structure: [sub_block1, comment1, sub_block2, comment2, ...]
+                else_block.append(line)
+
+            else:
+                sub_block.append(line)
+
+    # Add the last sub-block if it exists
+    if else_block:
+        # Sort the sub-blocks in the else block
+        else_block = [
+            [sorted(sub_block) if isinstance(sub_block, list) else sub_block for sub_block in sub_blocks]
+            for sub_blocks in else_block
+        ]
+        # Flatten the lists so they're all lines
+        else_block = [line for sub_block in else_block for line in sub_block]
+        new_init_file_lines.extend(else_block)
+
+    return "\n".join(new_init_file_lines)
+
+
+def update_init_file(filename, models):
+    with open(filename, "r") as f:
+        init_file = f.read()
+
+    # 1. For each model, find all the instances of model.model_name and replace with model.deprecated.model_name
+    for model in models:
+        init_file = init_file.replace(f"models.{model}", f"models.deprecated.{model}")
+
+    # 2. Resort the imports
+    init_file = update_alphabetic_ordering_of_imports(init_file)
+
+    with open(filename, "w") as f:
+        f.write(init_file)
+
+
+def remove_model_references_from_file(filename, models, condition):
+    """
+    Remove all references to the given models from the given file
+
+    Args:
+        filename (str): The file to remove the references from
+        models (List[str]): The models to remove
+        condition (Callable): A function that takes the line and model and returns True if the line should be removed
+    """
+    with open(filename, "r") as f:
+        init_file = f.read()
+
+    new_file_lines = []
+    for i, line in enumerate(init_file.split("\n")):
+        # stripped_line = line.strip().strip(",")
+        if any(condition(line, model) for model in models):
+            continue
+        new_file_lines.append(line)
+
+    with open(filename, "w") as f:
+        f.write("\n".join(new_file_lines))
+
+
+def remove_model_config_classes_from_config_check(filename, model_config_classes):
```

Comment:
I think this function only makes sense for `filename` being `utils/check_config_attributes.py`. So I am not very sure we need this `filename` argument.

Reply:
Also, it's better to make what types `model_config_classes` is expecting. I understand the items are string, but the name might make us think it expects `PretrainedConfig` objects. 

If arg name is kept as it is, nice to have a docstring.